SYCL/USM/alloc_functions.cpp

```diff
@@ -0,0 +1,246 @@
+  };
+  CheckAll(FAlign,
+           std::tuple{
+               [&]() { return ADevice(FAlign / 2, q); },
+               [&]() { return ADevice(FAlign / 2, d, ctx); },
+               [&]() { return ADevice(FAlign / 2, q, property_list{}); },
+               [&]() { return ADevice(FAlign / 2, d, ctx, property_list{}); }});
+  CheckAll(
+      Align,
+      std::tuple{[&]() { return ADevice(Align, q); },
+                 [&]() { return ADevice(Align, d, ctx); },
+                 [&]() { return ADevice(Align, q, property_list{}); },
+                 [&]() { return ADevice(Align, d, ctx, property_list{}); }});
+
+  auto AHost = [&](size_t Align, auto... args) {
+    return aligned_alloc_host(Align, sizeof(std::max_align_t), args...);
+  };
+  CheckAll(
+      FAlign,
+      std::tuple{[&]() { return AHost(FAlign / 2, q); },
+                 [&]() { return AHost(FAlign / 2, ctx); },
+                 [&]() { return AHost(FAlign / 2, q, property_list{}); },
+                 [&]() { return AHost(FAlign / 2, ctx, property_list{}); }});
+  CheckAll(Align,
+           std::tuple{[&]() { return AHost(Align, q); },
+                      [&]() { return AHost(Align, ctx); },
+                      [&]() { return AHost(Align, q, property_list{}); },
+                      [&]() { return AHost(Align, ctx, property_list{}); }});
+
+  auto AShared = [&](size_t Align, auto... args) {
+    return aligned_alloc_shared(Align, sizeof(std::max_align_t), args...);
+  };
+  CheckAll(FAlign,
+           std::tuple{
+               [&]() { return AShared(FAlign / 2, q); },
+               [&]() { return AShared(FAlign / 2, d, ctx); },
+               [&]() { return AShared(FAlign / 2, q, property_list{}); },
+               [&]() { return AShared(FAlign / 2, d, ctx, property_list{}); }});
+  CheckAll(
+      Align,
+      std::tuple{[&]() { return AShared(Align, q); },
+                 [&]() { return AShared(Align, d, ctx); },
+                 [&]() { return AShared(Align, q, property_list{}); },
+                 [&]() { return AShared(Align, d, ctx, property_list{}); }});
+
+  auto TDevice = [&](auto... args) {
+    return malloc_device<Aligned>(1, args...);
+  };
+  CheckAll(Align, std::tuple{[&]() { return TDevice(q); },
+                             [&]() { return TDevice(d, ctx); }});
+
+  auto THost = [&](auto... args) { return malloc_host<Aligned>(1, args...); };
+  CheckAll(Align, std::tuple{[&]() { return THost(q); },
+                             [&]() { return THost(ctx); }});
+
+  auto TShared = [&](auto... args) {
+    return malloc_shared<Aligned>(1, args...);
+  };
+  CheckAll(Align, std::tuple{[&]() { return TShared(q); },
+                             [&]() { return TShared(d, ctx); }});
+
+  auto ATDevice = [&](size_t Align, auto... args) {
+    return aligned_alloc_device<Aligned>(Align, 1, args...);
+  };
+  CheckAll(Align, std::tuple{[&]() { return ATDevice(Align / 2, q); },
```

Comment:
At this place and few other as well: why do you request alignment = "Align/2" and expect the actual alignment would be twice bigger, i.e. "Align" ?

Reply:
From table 94 in https://github.com/gmlueck/SYCL-Docs/blob/8c37a7eb9483938f35cace000c11be12ea77538e/adoc/chapters/programming_interface.adoc#usm-allocations:

<html>
<body>
<!--StartFragment-->

Alignment parameter alignment specified Templated on allocation type T | Pointer is suitably aligned for an object of type T or it is aligned to the specified alignment, whichever is greater.
-- | --


<!--EndFragment-->
</body>
</html>

The `Align / 2` is less than the template type parameter's alignment so the latter must be used.

Reply:
I see now, thank you for the link.